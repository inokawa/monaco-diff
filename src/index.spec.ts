import { describe, it, expect } from "@jest/globals";
import { diff, legacyLinesDiff, linesDiff } from "../lib";

describe("computeDiff", () => {
  it("singleline-1", () => {
    expect(diff(["heLLo world!"], ["hello orlando!"])).toMatchSnapshot();
  });
  it("multiline-1", () => {
    expect(
      diff(
        "just some text\n\nHello World\n\nSome more text".split("\n"),
        "just some Text\n\nHello World\n\nSome more changes".split("\n")
      )
    ).toMatchSnapshot();
  });
  it("multiline-2", () => {
    expect(
      diff(
        "This line is removed on the right.\njust some text\nabcd\nefgh\nSome more text".split(
          "\n"
        ),
        "just some text\nabcz\nzzzzefgh\nSome more text.\nThis line is removed on the left.".split(
          "\n"
        )
      )
    ).toMatchSnapshot();
  });
});

describe("compute legacy line diff", () => {
  it("singleline-1", () => {
    expect(
      legacyLinesDiff(["heLLo world!"], ["hello orlando!"])
    ).toMatchSnapshot();
  });
  it("multiline-1", () => {
    expect(
      legacyLinesDiff(
        "just some text\n\nHello World\n\nSome more text".split("\n"),
        "just some Text\n\nHello World\n\nSome more changes".split("\n")
      )
    ).toMatchSnapshot();
  });
  it("multiline-2", () => {
    expect(
      legacyLinesDiff(
        "This line is removed on the right.\njust some text\nabcd\nefgh\nSome more text".split(
          "\n"
        ),
        "just some text\nabcz\nzzzzefgh\nSome more text.\nThis line is removed on the left.".split(
          "\n"
        )
      )
    ).toMatchSnapshot();
  });
});

describe("compute line diff", () => {
  it("singleline-1", () => {
    expect(linesDiff(["heLLo world!"], ["hello orlando!"])).toMatchSnapshot();
  });
  it("multiline-1", () => {
    expect(
      linesDiff(
        "just some text\n\nHello World\n\nSome more text".split("\n"),
        "just some Text\n\nHello World\n\nSome more changes".split("\n")
      )
    ).toMatchSnapshot();
  });
  it("multiline-2", () => {
    expect(
      linesDiff(
        "This line is removed on the right.\njust some text\nabcd\nefgh\nSome more text".split(
          "\n"
        ),
        "just some text\nabcz\nzzzzefgh\nSome more text.\nThis line is removed on the left.".split(
          "\n"
        )
      )
    ).toMatchSnapshot();
  });
});
