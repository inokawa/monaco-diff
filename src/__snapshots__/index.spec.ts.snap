// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`computeDiff multiline-1 1`] = `
[
  LineChange {
    "charChanges": [
      CharChange {
        "modifiedEndColumn": 12,
        "modifiedEndLineNumber": 1,
        "modifiedStartColumn": 11,
        "modifiedStartLineNumber": 1,
        "originalEndColumn": 12,
        "originalEndLineNumber": 1,
        "originalStartColumn": 11,
        "originalStartLineNumber": 1,
      },
    ],
    "modifiedEndLineNumber": 1,
    "modifiedStartLineNumber": 1,
    "originalEndLineNumber": 1,
    "originalStartLineNumber": 1,
  },
  LineChange {
    "charChanges": [
      CharChange {
        "modifiedEndColumn": 18,
        "modifiedEndLineNumber": 5,
        "modifiedStartColumn": 11,
        "modifiedStartLineNumber": 5,
        "originalEndColumn": 15,
        "originalEndLineNumber": 5,
        "originalStartColumn": 11,
        "originalStartLineNumber": 5,
      },
    ],
    "modifiedEndLineNumber": 5,
    "modifiedStartLineNumber": 5,
    "originalEndLineNumber": 5,
    "originalStartLineNumber": 5,
  },
]
`;

exports[`computeDiff multiline-2 1`] = `
[
  LineChange {
    "charChanges": undefined,
    "modifiedEndLineNumber": 0,
    "modifiedStartLineNumber": 0,
    "originalEndLineNumber": 1,
    "originalStartLineNumber": 1,
  },
  LineChange {
    "charChanges": [
      CharChange {
        "modifiedEndColumn": 5,
        "modifiedEndLineNumber": 3,
        "modifiedStartColumn": 4,
        "modifiedStartLineNumber": 2,
        "originalEndColumn": 5,
        "originalEndLineNumber": 3,
        "originalStartColumn": 4,
        "originalStartLineNumber": 3,
      },
      CharChange {
        "modifiedEndColumn": 34,
        "modifiedEndLineNumber": 5,
        "modifiedStartColumn": 15,
        "modifiedStartLineNumber": 4,
        "originalEndColumn": 15,
        "originalEndLineNumber": 5,
        "originalStartColumn": 15,
        "originalStartLineNumber": 5,
      },
    ],
    "modifiedEndLineNumber": 5,
    "modifiedStartLineNumber": 2,
    "originalEndLineNumber": 5,
    "originalStartLineNumber": 3,
  },
]
`;

exports[`computeDiff singleline-1 1`] = `
[
  LineChange {
    "charChanges": [
      CharChange {
        "modifiedEndColumn": 7,
        "modifiedEndLineNumber": 1,
        "modifiedStartColumn": 3,
        "modifiedStartLineNumber": 1,
        "originalEndColumn": 8,
        "originalEndLineNumber": 1,
        "originalStartColumn": 3,
        "originalStartLineNumber": 1,
      },
      CharChange {
        "modifiedEndColumn": 14,
        "modifiedEndLineNumber": 1,
        "modifiedStartColumn": 10,
        "modifiedStartLineNumber": 1,
        "originalEndColumn": 12,
        "originalEndLineNumber": 1,
        "originalStartColumn": 11,
        "originalStartLineNumber": 1,
      },
    ],
    "modifiedEndLineNumber": 1,
    "modifiedStartLineNumber": 1,
    "originalEndLineNumber": 1,
    "originalStartLineNumber": 1,
  },
]
`;
